v3/networks.proto

```diff
@@ -91,3 +95,28 @@ message NetworkAPIKey {
   // Secret key value.
   string key = 6;
 }
+
+message Target {
+  // Protocol to use.
+  TargetProtocol protocol = 1;
+  // Pointer to the target.
+  // When using an HTTP protocol, this can be a URL with scheme, host, port and path.
+  // This value may be empty if the protocol supports service lookup (e.g. via DNS).
+  string value = 2;
+
+  message BasicAuth {
+    // Username.
+    string username = 1;
+    // Password.
+    string password = 2;
+  }
+  message CustomAuth {
+    // Custom authorization value.
+    // When using an HTTP protocol, this is used as the Authorization header value.
```

Comment:
Maybe add a small hint that this should also include the authentication type, such as `Bearer`.